src/ParquetMsg.chpl

```diff
@@ -122,13 +122,34 @@ module ParquetMsg {
     }
   }
 
-  proc getStrColSize(filename: string, dsetname: string) throws {
-    extern proc c_getStringColumnNumBytes(filename, colname, errMsg): int;
+  proc calcSizesAndOffset(offsets: [] ?t, byteSizes: [] int, filenames: [] string, sizes: [] int, dsetname: string) throws {
+    var (subdoms, length) = getSubdomains(sizes);
+    
+    coforall loc in offsets.targetLocales() do on loc {
+      var locFiles = filenames;
+      var locFiledoms = subdoms;
+      forall (i, filedom, filename) in zip(sizes.domain, locFiledoms, locFiles) {
```

Comment:
Yes, I think you've got it right, there shouldn't be duplicated work because the `forall` is with default rectangular arrays on `loc`.

If you look closely, you will see that the column that we are using here is declared as `var col: [filedom] t;`, when it should ideally be defined as `var col: [intersection] t;` and then only read a subset of the data, rather than all of the contents of the file, but due to how we have implemented the Parquet stuff up til today, we are unfortunately reading in entire files even if the intersection on a single locale is only, say, a single element.

I hate to keep saying we have plans to resolve this soon, but this is another thing we plan to resolve soon. We have been investigating ways to only read the necessary portions of the file, and think we have figured out the path forward, which is to skip to the `intersection.low` in the Parquet file and assign only the necessary amount of elements directly into the local slice of the distributed array, which will also avoid the extra copy that `col` is introducing here and in the main `readFilesByName` loop.

Reply:
Ah, that's a good point. It's totally fine that this is on the "plan to resolve soon" list :)